### ISSUE TITLE
Using ">" in shell mode is dangerous

### ISSUE THREAD

**Issue by diegozea** [JuliaLang/julia#24529]
Doing something like `grep ">" file` this deletes the content of the file using the julia shell mode.

```
dzea@slim:~$ cat example.fasta 
>Seq1
MELRTPGTKWSPQGI
>Seq_2
MELRTPCRVHG
dzea@slim:~$ grep ">" example.fasta 
>Seq1
>Seq_2
dzea@slim:~$ julia06
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: https://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.6.1 (2017-10-24 22:15 UTC)
 _/ |\__'_|_|_|\__'_|  |  Official http://julialang.org/ release
|__/                   |  x86_64-pc-linux-gnu

shell> cat example.fasta
>Seq1
MELRTPGTKWSPQGI
>Seq_2
MELRTPCRVHG

shell> grep ">" example.fasta
Utilisation : grep [OPTION]... MOTIF [FICHIER]...
Exécutez « grep --help » pour obtenir des renseignements complémentaires.

shell> cat example.fasta

julia> versioninfo()
Julia Version 0.6.1
Commit 0d7248e2ff (2017-10-24 22:15 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Xeon(R) CPU           E5620  @ 2.40GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Nehalem)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.9.1 (ORCJIT, westmere)
```

**Comment by diegozea** [comment]
It's only a problem in shell mode. It works fine using `run`:
```
shell> cat example.fasta
>Seq1
MELRTPGTKWSPQGI
>Seq_2
MELRTPCRVHG

julia> run(`grep ">" example.fasta`)
>Seq1
>Seq_2

shell> cat example.fasta
>Seq1
MELRTPGTKWSPQGI
>Seq_2
MELRTPCRVHG

```

**Comment by vtjnash** [comment]
dup #10120. fixed by #23881

**Comment by JeffBezanson** [comment]
Fixed.
